### PR TITLE
Add permissions

### PR DESCRIPTION
**Pull request by eyalvo-matrix**
Hi 
This is a security permission fix for the codes screen 
(We need to consider applying additional fixes to other screens related to the  superbill  permission )

Today users that have superbill permission can edit the data in the codes screen
regardless to the view/edit state of the permission!
Now after the fix ,  users that have superbill view permission can not edit the data in that screen

Thanks
Eyal

### CODE REVIEW

File: interface/patient_file/encounter/superbill_custom_full.php [hunk 1]
```diff
@@ -11,20 +11,31 @@
 require_once("../../../custom/code_types.inc.php");
 require_once("$srcdir/options.inc.php");
 
+// gacl control
+$thisauthview = acl_check('admin', 'superbill', false, 'view');
+$thisauthwrite = acl_check('admin', 'superbill', false, 'write');
+
+if (!($thisauthwrite || $thisauthview)) {
+    echo "<html>\n<body>\n";
+    echo "<p>".xl('You are not authorized for this.','','','</p>')."\n";
```
Comment (bradymiller) [inline interface/patient_file/encounter/superbill_custom_full.php]: Do this instead for above line:
``````
echo "<p>" . xlt('You are not authorized for this.') . "</p>\n";

File: interface/patient_file/encounter/superbill_custom_full.php [hunk 2]
```diff
@@ -36,179 +47,179 @@ function bucks($amount) {
...
-  $cyp_factor = $_POST['cyp_factor'] + 0;
-  $active     = empty($_POST['active']) ? 0 : 1;
-  $reportable = empty($_POST['reportable']) ? 0 : 1; // dx reporting
-  $financial_reporting = empty($_POST['financial_reporting']) ? 0 : 1; // financial service reporting
-
-  $taxrates = "";
-  if (!empty($_POST['taxrate'])) {
-    foreach ($_POST['taxrate'] as $key => $value) {
-      $taxrates .= "$key:";
+if (isset($mode)&&$thisauthwrite) {
```
Comment (bradymiller) [inline interface/patient_file/encounter/superbill_custom_full.php]: needs some spacing:
``````
if (isset($mode) && $thisauthwrite) {

File: interface/patient_file/summary/demographics.php [hunk 3]
```diff
@@ -1306,7 +1306,7 @@ function setMyPatient() {
     // Show Clinical Reminders for any user that has rules that are permitted.
     $clin_rem_check = resolve_rules_sql('','0',TRUE,'',$_SESSION['authUser']);
     if (!empty($clin_rem_check) && $GLOBALS['enable_cdr'] && $GLOBALS['enable_cdr_crw'] &&
-        acl_check('patients', 'reminder'))
```
Comment (matrix-amiel) [inline interface/patient_file/summary/demographics.php]: I think it mistake.. it should be 'alert' ACO for clinical reminders widget.
image

File: interface/patient_file/encounter/superbill_custom_full.php (hunk 1 above)
Comment (eyalvo-matrix) [inline interface/patient_file/encounter/superbill_custom_full.php]: Thanks you are right it is better

File: interface/patient_file/encounter/superbill_custom_full.php (hunk 2 above)
Comment (eyalvo-matrix) [inline interface/patient_file/encounter/superbill_custom_full.php]: Ok

File: interface/patient_file/summary/demographics.php (hunk 3 above)
Comment (eyalvo-matrix) [inline interface/patient_file/summary/demographics.php]: This is an additional fix from my work colleague

**Comment by bradymiller** [comment]
very straightforward, so will bring into codebase. thanks! -brady

**Pull request merged**